ecommerce/views.py

```diff
@@ -36,3 +46,46 @@ def post(self, request, *args, **kwargs):
             'payload': payload,
             'url': settings.CYBERSOURCE_SECURE_ACCEPTANCE_URL,
         })
+
+
+class OrderFulfillmentView(APIView):
+    """
+    View for order fulfillment API. This API is special in that only CyberSource should talk to it.
+    Instead of authenticating with OAuth or via session this looks at the signature of the message
+    to verify authenticity.
+    """
+
+    authentication_classes = ()
+    permission_classes = (IsSignedByCyberSource, )
+
+    def post(self, request, *args, **kwargs):
+        """
+        Confirmation from CyberSource which fulfills an existing Order.
+        """
+        # First, save this information in a receipt
+        receipt = Receipt.objects.create(data=request.data)
+
+        # Link the order with the receipt if we can parse it
+        reference_number = request.data['req_reference_number']
+        order = get_new_order_by_reference_number(reference_number)
+        receipt.order = order
+        receipt.save()
+
+        try:
+            if request.data['decision'] != 'ACCEPT':
+                # This may happen if the user clicks 'Cancel Order'
+
+                order.status = Order.FAILED
+                order.save()
+            else:
+                # Do the verified enrollment with edX here
+
+                order.status = Order.FULFILLED
+                order.save()
```

Comment:
line breaks at 77 and 82 don't seem necessary, and `order.save()` can be moved outside of the if/else instead of repeated in both conditions